packages/expo-manifests/ios/EXManifests/EXManifestsBaseManifest.m

```diff
@@ -305,4 +310,14 @@ + (NSString * _Nullable)getStringFromManifest:(NSDictionary *)manifest
   return nil;
 }
 
+- (int)sdkMajorVersion
+{
+  NSString *sdkVersion = [self.expoClientConfigRootObject nullableStringForKey:@"sdkVersion"];
```

Comment:
this should also call `sdkVersion`?

Reply:
(this will break for non-classic updates since SDK version is defined differently)

Reply:
thanks @wschurman for catching up on this. https://github.com/expo/expo/pull/21300 should address the problem. i also tested expo go without #21300 and it works for these cases as well:
  - local sdk 48 app
  - eas update with `runtime=1.0.0`
    - expo go home does not load this app, which is expected.
  - eas update with `runtime.policy=sdkVersion`
    - that works because there is a sdkVersion inside expoClient, i.e. `manifest.extra.expoClient.sdkVersion=48.0.0` even though the `runtimeVersion=exposdk:48.0.0`.

i was testing that to make sure it works fine on expo go and we don't have to resubmit a new expo go to app store. #21300 should be better than the current status quo, we could publish the fix in new expo-manifests to make sure it does not break development builds.

> this should also call sdkVersion method? (like android implementation)

that is not like the android's sdkVersion, it's the [sdkMajorVersion on android](https://github.com/expo/expo/blob/df226942495171121026668a8713bdb89bf005ac/packages/expo-manifests/android/src/main/java/expo/modules/manifests/core/Manifest.kt#L176). simply because objective-c does not support those inline syntax sugars 😅 we could refactor this after your #21298 landed

Reply:
> eas update with `runtime.policy=sdkVersion`
> * that works because there is a sdkVersion inside expoClient, i.e. `manifest.extra.expoClient.sdkVersion=48.0.0` even though the `runtimeVersion=exposdk:48.0.0`.

I think that this field is no longer required? (sdk version is derived from expo package version in package.json)

Reply:
right, [it gets the version from the expo package's package.json](https://github.com/expo/expo/blob/13e0d56d983b700b8c12ba5cd408dcd063d5aa3a/packages/%40expo/config/src/getExpoSDKVersion.ts#L22-L43). just tested `eas update` when no **node_modules/expo** . it shows the error:

```
    ConfigError: Cannot determine which native SDK version your project uses because the module `expo` is not installed. Please install it with `yarn add expo` and try again.
    Code: MODULE_NOT_FOUND
```
it should be because eas-cli doesn't `skipSDKVersionRequirement`.
so i think we could still be confident about reading `manifest.extra.expoClient.sdkVersion` in current expo go.